public/src/components/ToastManager.jsx

```diff
@@ -0,0 +1,198 @@
+      transform: translateY(0px)
+    }
+  }
+  @keyframes cancelled {
+  0% {
+    transform: translateX(0px);
+  }
+  90% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+  @keyframes immediatecancel {
+  0% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+
+  display: flex;
+  border: 1px solid #54457F;
+  width: 250px;
+  cursor: pointer;
+  animation: styledtoast .3s ease-out, cancelled 4.5s ease-out .3s;
+  animation-fill-mode: forwards;  
+  border-radius: .25em;
+  background-color: white;
+  box-shadow: 0px 4px 3px 3px rgba(0,0,0, .1);
+  & + & {
+    margin-top: 1em;
+  }
+
+  & .toast__alert {
+    margin-left: -1px;
+    border-radius: .25em 0 0 .25em;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 25px;
+    & path {
+      fill: white;
+    }
+    padding: .25em .5em;
+    ${(props) => getCSSForFlavor(props.flavor)}
+  }
+  & .toast__content {
+    padding: .5em 1em;
+    flex: 1;
+  }
+  &.toast--cancelled {
+    animation: immediatecancel .35s ease-out;
+  }
+  & .toast__cancel {
+    display: flex;
+    padding: .25em .5em;
+    & path {
+      fill: #54457F;
+    }
+    align-items: center;
+  }
+`;
+
+export const ToastAlert = ({ children, flavor, id }) => {
+  const { manuallyCancel, flavors } = React.useContext(ToastContext);
+  const internalFlavors = {
+    [flavors.success]: <FontAwesomeIcon icon={faCheckCircle} />,
+    [flavor.error]: <FontAwesomeIcon icon={faExclamationTriangle} />,
+  };
+  const alertIcon = internalFlavors[flavor];
+  const [cancel, setCancel] = useState(false);
+  const handleClick = () => {
+    setCancel(true);
+    manuallyCancel(id);
+  };
+  return (
+    <StyledToast
+      flavor={flavor}
+      ariaLive="polite"
+      className={cancel ? 'toast--cancelled' : ''}
+      onClick={handleClick}
+    >
+      <div className="toast__alert">
+        {alertIcon}
+      </div>
+      <div className="toast__content">
+        {children}
+      </div>
+      <div className="toast__cancel">
+        <FontAwesomeIcon icon={faTimes} />
+      </div>
+    </StyledToast>
+  );
+};
+
+ToastAlert.propTypes = {
+  children: PropTypes.node.isRequired,
+  flavor: PropTypes.string.isRequired,
+  id: PropTypes.number.isRequired,
+};
+
+export const ToastManager = ({ children, defaultTime }) => {
+  const toasts = useRef([]);
+  const [availableToastId, setAvailableToastId] = useState(0);
+  const [forceRerender, setForceRerender] = useState(0);
+
+  const contextData = {
+    flavors: {
+      success: 'success',
+      error: 'error',
+    },
+    toast(message, flavor, time) {
+      const usedId = availableToastId;
+      toasts.current = [...toasts.current, {
+        message, flavor, time, id: usedId,
+      }];
+
+      setAvailableToastId(availableToastId + 1);
+      setTimeout(() => {
+        toasts.current.splice(0, 1); // toasts.filter((toast) => toast.id !== usedId);
```

Comment:
remove unused code instead of commenting

Reply:
Fixed.